### ISSUE TITLE
Style/MultilineWhenThen detection failure

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#8158]
The following:

```ruby
# in lib/rubocop/ast/node_pattern.rb

         def compile_arg(token)
           case token
           when WILDCARD  then
             name = token[1..-1]
             access_unify(name) || fail_due_to('invalid in arglist: ' + token)
           when LITERAL   then token
          # ...
```

does not generate an offense (it should). I realized that as I refactored it as:

```ruby
         def compile_arg(token)
           name = token[1..-1]
           case token
           when WILDCARD  then
             access_unify(name) || fail_due_to('invalid in arglist: ' + token)
           when LITERAL   then token
          # ...
```

which generates an offense (correct):
```
lib/rubocop/ast/node_pattern.rb:656:26: C: [Corrected] Style/MultilineWhenThen: Do not use then for multiline when statement.
          when WILDCARD  then
                         ^^^^
```
